### PR TITLE
PP 63 approve reject application front 

### PR DESCRIPTION
**Pull request by mariyador**
### Description

This update introduces functions to approve and reject applications within the application.

### Changes

State Management:
- handleApprove and handleDecline functions update the state by filtering out the application that has been approved or rejected and removing classes with no remaining applications.
- Both functions also update the list of applicants to remove the corresponding applicant after approval or rejection.

Error Handling:
- Added error handling in both functions to catch and display errors in case the approval or rejection process fails.

API Interaction:
- Implemented approveApplication and rejectApplication functions to send POST requests to the API for approving or rejecting an application, respectively.

### CODE REVIEW

**Comment by vrudnitskaya** [comment]
Well done, Maria! I was able to approve and reject several applications. Please add styles to the information about no applications.
no-applications

**Comment by mariyador** [comment]
> Well done, Maria! I was able to approve and reject several applications. Please add styles to the information about no applications. no-applications

Thank you, Valya!
Update the style of application page if no applications:
Screenshot 2024-09-01 at 12 54 38 AM

**Review by vrudnitskaya — APPROVED** [review]

File: src/components/pages/dashboard-pages/Notifications.jsx [hunk 1]
```diff
@@ -1,10 +1,284 @@
...
+      }
+    };
+
+    getStudentInfo();
+  }, [classes, userData]);
+
+  const handleApprove = async (classId, applicationId) => {
+    const token = userData.token;
+    try {
+      const response = await approveApplication(token, classId, applicationId);
```
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/Notifications.jsx:94]: we have unused variable response here

File: src/components/pages/dashboard-pages/Notifications.jsx [hunk 2]
```diff
@@ -1,10 +1,284 @@
...
+      setApplicantsError({
+        message: 'Failed to approve application. Please try again later.',
+      });
+    }
+  };
+
+  const handleDecline = async (classId, applicationId) => {
+    const token = userData.token;
+    try {
+      const response = await rejectApplication(token, classId, applicationId);
```
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/Notifications.jsx:123]: variable response unused here too

File: src/components/pages/dashboard-pages/Notifications.jsx [hunk 3]
```diff
@@ -1,10 +1,284 @@
...
+  }
+
   return (
-    <>
-      <h1 className="text-red font-bold text-2xl sm:text-4xl font-spartan uppercase">
-        Notifications page - common
-      </h1>
-    </>
+    <div className="container mt-10">
+      {classesError && <p className="text-red-500">{classesError.message}</p>}
```
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/Notifications.jsx:155]: text-red was changed on red-500, is it intentionally?

**Review by AnnaPestova1 — COMMENTED** [review]
Great functionality! Please check colors (see PR message)

File: src/components/pages/dashboard-pages/Notifications.jsx [hunk 4]
```diff
@@ -1,10 +1,284 @@
...
+                );
+              })}
+            </div>
+          ))
+        ) : (
+          <div className="text-center mt-10">
+            <h1 className="text-2xl font-bold text-gray-800 mb-4">
+              No Applications for Your Classes
+            </h1>
+            <p className="text-lg text-gray-600 mb-6">
```
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/Notifications.jsx:258]: check colors please

File: src/components/pages/dashboard-pages/Notifications.jsx (hunk 1 above)
Comment (mariyador) [reply src/components/pages/dashboard-pages/Notifications.jsx:94]: Thank you!
Fixed

File: src/components/pages/dashboard-pages/Notifications.jsx (hunk 2 above)
Comment (mariyador) [reply src/components/pages/dashboard-pages/Notifications.jsx:123]: Fixed

File: src/components/pages/dashboard-pages/Notifications.jsx (hunk 4 above)
Comment (mariyador) [reply src/components/pages/dashboard-pages/Notifications.jsx]: ✅

**Review by AnnaPestova1 — APPROVED** [review]
Thank you!

**Pull request merged**